usr/share/rear/layout/prepare/default/420_autoresize_last_partitions.sh

```diff
@@ -0,0 +1,267 @@
+#    or where its GPT name or flags contains 'efi' or 'esp' (anywhere case insensitive)
+# are excluded from being resized e.g. as in
+# AUTORESIZE_EXCLUDE_PARTITIONS=( boot swap efi /dev/sda2 /dev/sdb3 )
+#
+# The last active partition on each active disk gets resized but nothing more.
+# In particular this does not resize volumes on top of the affected partitions.
+# To migrate volumes on disk where the disk size had changed the user must in advance
+# manually adapt his disklayout.conf file before he runs "rear recover".
+#
+# In general ReaR is not meant to somehow "optimize" a system during "rear recover".
+# ReaR is meant to recreate a system as much as possible exactly as it was before.
+# Accordingly this automated resizing implements a "minimal changes" approach:
+#
+# When the new disk is a bit smaller (at most AUTOSHRINK_DISK_SIZE_LIMIT_PERCENTAGE percent),
+# only the last (active) partition gets shrinked but all other partitions are not changed.
+# When the new disk is smaller than AUTOSHRINK_DISK_SIZE_LIMIT_PERCENTAGE percent it errors out.
+# To migrate onto a substantially smaller new disk the user must in advance
+# manually adapt his disklayout.conf file before he runs "rear recover".
+#
+# When the new disk is only a bit bigger (less than AUTOINCREASE_DISK_SIZE_THRESHOLD_PERCENTAGE percent),
+# no partition gets increased (which leaves the bigger disk space at the end of the disk unused).
+# When the new disk is substantially bigger (at least AUTOINCREASE_DISK_SIZE_THRESHOLD_PERCENTAGE percent),
+# only the last (active) partition gets increased but all other partitions are not changed.
+# To migrate various partitions onto a substantially bigger new disk the user must in advance
+# manually adapt his disklayout.conf file before he runs "rear recover".
+#
+# Because only the end value of the last partition may get changed, the partitioning alignment
+# of the original system is not changed, cf. https://github.com/rear/rear/issues/102
+#
+# Because only the last active (i.e. not commented in disklayout.conf) partition on a disk
+# may get changed, things go wrong if another partition is actually the last one on the disk
+# but that other partition is commented in disklayout.conf (e.g. because that partition
+# is a partition of another operating system that is not mounted during "rear mkrescue").
+# To migrate a system with a non-active last partition onto a bigger or smaller new disk
+# the user must in advance manually adapt his disklayout.conf file before he runs "rear recover".
+
+# Skip if not in migration mode:
+is_true "$MIGRATION_MODE" || return 0
+
+# Skip if automatically resize partitions is explicity unwanted:
+is_false "$AUTORESIZE_PARTITIONS" && return 0
+
+# Skip resizing only the last partition if resizing all partitions is explicity wanted
+# which is done by the separated 430_autoresize_all_partitions.sh script:
+is_true "$AUTORESIZE_PARTITIONS" && return 0
```

Comment:
@jsmeix is `is_false "$AUTORESIZE_PARTITIONS" && return 0` not the opposite from the previous statement?

Reply:
@gdha 
your usual question ;-))
and my usual answer: No - because "tertium datur":
it_true and is_falseare not logical opposites,
see their documenation in their function definition code
or see the "Relax-and-Recover functions"section in
https://github.com/rear/rear/wiki/Coding-Style